spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/SpringSecurityAuditorAwareTests.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.service;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.cloud.common.security.support.SecurityStateBean;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.server.resource.authentication.BearerTokenAuthentication;
+import org.springframework.security.oauth2.server.resource.introspection.OAuth2IntrospectionAuthenticatedPrincipal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for org.springframework.cloud.dataflow.server.service.SpringSecurityAuditorAware
+ * @author Corneil du Plessis
+ */
+public class SpringSecurityAuditorAwareTests {
+	/**
+	 * Test the fix for the case where the authentication doesn't contain a name but isn't anonymous it resulted in an NPE.
+	 */
+	@Test
+	public void testNullPointerOnUnnamedPrincipal() {
```

Comment:
**[NIT]** 2 features Junit5 brings that we should use more is 
1) no need for public class or methods
2) no need to prefix method names w/ "test"

I will adjust during merge.